### PR TITLE
Replace sun-moon icon

### PR DESCRIPTION
**Pull request by itsjavi**
## What is the purpose of this pull request?

- [x] New Icon
- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other:

### Description

This version of the sun-moon icon is an alternate version with the moon + the shining lines of the sun.
You can see more clearly that is a moon, instead of 2 combined circles from the other version.

This icon represents a combination of the moon and sun icons more clearly.

image

Use cases: when you want to represent "OS preference" when it comes to dark-mode theme on a website.

### Concept 

- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license

- [ ] The icons are solely my own creation.
- [ ] The icons were originally created in #<issueNumber> by @<githubUser>
- [x] I've based them on the following Lucide icons: sun, moon, sun-moon
- [ ] I've based them on the following design: 

### Naming 

- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design 

- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to two points of precision.

## Before Submitting 

- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

File: icons/sun-moon-2.svg [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+>
+  <path d="M12 8a2.83 2.83 0 0 0 4 4c0 2.2-1.8 4-4 4s-4-1.8-4-4 1.8-4 4-4z"/>
+  <path d="M12 2v2"/>
+  <path d="M12 20v2"/>
+  <path d="m4.93 4.93 1.41 1.41"/>
+  <path d="m17.66 17.66 1.41 1.41"/>
+  <path d="M2 12h2"/>
+  <path d="M20 12h2"/>
+  <path d="m6.34 17.66-1.41 1.41"/>
+  <path d="m19.07 4.93-1.41 1.41"/>
```
Comment (jguddas) [inline icons/sun-moon-2.svg:20]:
```suggestion
  <path d="M12 8a2.83 2.83 0 0 0 4 4 4 4 0 1 1-4-4" />
  <path d="M12 2v2" />
  <path d="M12 20v2" />
  <path d="m4.9 4.9 1.4 1.4" />
  <path d="m17.7 17.7 1.4 1.4" />
  <path d="M2 12h2" />
  <path d="M20 12h2" />
  <path d="m6.3 17.7-1.4 1.4" />
  <path d="m19.1 4.9-1.4 1.4" />
```

File: icons/sun-moon-2.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "itsjavi",
+    "mittalyashu",
+    "karsa-mistmere"
+  ],
+  "tags": [
+    "night",
+    "light",
+    "moon",
+    "sun",
+    "brightness"
```
Comment (jguddas) [inline icons/sun-moon-2.json:13]: Would be awesome to add some use case related tags here.

**Review by jguddas — COMMENTED** [review]
Looks great, we don't add `-2` icons, but I think replacing `sun-moon` with your version is a good idea.

**Review by jguddas — APPROVED** [review]

**Review by karsa-mistmere — APPROVED** [review]

**Review by ericfennis — CHANGES_REQUESTED** [review]

**Review by ericfennis — APPROVED** [review]
Nice work, good to go!

**Pull request merged**